src/monodroid/jni/android-system.cc

```diff
@@ -226,6 +226,24 @@ AndroidSystem::monodroid_get_system_property (const char *name, char **value)
 	return len;
 }
 
+int
+AndroidSystem::monodroid_read_file_into_memory (const char *path, char **value)
+{
+	int r = 0;
```

Comment:
I think we should insert:

```cc
if (value) {
    *value = NULL;
}
```

so that if the path cannot be opened, `*value` isn't potentially left as garbage.